packages/js/data/src/products/test/reducer.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * Internal dependencies
+ */
+import reducer, { ProductState } from '../reducer';
+import TYPES from '../action-types';
+import {
+	getProductResourceName,
+	getTotalProductCountResourceName,
+} from '../utils';
+import { Actions } from '../actions';
+import { PartialProduct, ProductQuery } from '../types';
+
+const defaultState: ProductState = {
+	products: {},
+	productsCount: {},
+	errors: {},
+	data: {},
+};
+
+describe( 'products reducer', () => {
+	it( 'should return a default state', () => {
+		const state = reducer( undefined, {} as Actions );
+		expect( state ).toEqual( defaultState );
+		expect( state ).not.toBe( defaultState );
+	} );
+
+	it( 'should handle GET_PRODUCT_SUCCESS', () => {
+		const itemType = 'guyisms';
+		const initialState: ProductState = {
+			products: {
+				[ itemType ]: {
+					data: [ 1, 2 ],
+				},
+			},
+			productsCount: {
+				'total-guyisms:{}': 2,
```

Comment:
😆

Reply:
Lol, this was deff not me haha I copy pasted most of these tests and updated them, I somehow never noticed this.